ezmlm-browse-setup.1
.TH ezmlm-browse-conf 1
.SH NAME
setup \- what it does
.SH SYNOPSIS
.B ezmlm-browse-conf
[
.B \-\-apache
] [
.B \-\-setuid
]
.I DESTPATH TARGET.cgi [ LISTPATH HOSTNAME ]
.SH DESCRIPTION
This program sets up the necessary links and programs to use
.B ezmlm-browse
on a web server.
.P
.I DESTPATH
is the path into which the files should be installed.  This is
necessarily a directory which the web server can access.
.I TARGET.cgi
is the filename for the CGI script in the target directory.  This will
frequently be
.I index.cgi
if there is nothing else in the directory.  A sample configuration file
is written to the destination directory if it is not already present.
.SH OPTIONS
.TP
.B \-\-apache
Create an Apache
.B .htaccess
configuration file in the destination.
.TP
.B \-\-setuid
Install a setuid CGI wrapper in the destination (instead of the usual
startup script) in order to allow
.B ezmlm-browse
to read the archive files.  This is useful if the archive files are
owned by a different user than the web server and Apache suexec (or
equivalent) is not in use.


NEWS
-------------------------------------------------------------------------------
Changes in version 0.21

Development of this version has been sponsored by FutureQuest, Inc.
ossi@FutureQuest.net  http://www.FutureQuest.net/
-------------------------------------------------------------------------------
Changes in version 0.20

- Ezmlm-browse now properly converts all messages to UTF-8 on output.

- Ezmlm-browse is now set up to build to and install as a single file.

- Switched configuration to use a ezmlm-browse.ini file.  An
  ezmlm-browse-reconfig program is included to generate this .ini file
  from an existing config.py file.

- Added an ezmlm-browse-setup script to simplify installation.

- Fixed handling of messages with no date stamp.

- Default templates now show attached images inline.

- Downloading of images is controlled separately from downloading the
  whole raw message.

- Fixed broken HTML in pager and menu bar links.

Development of this version has been sponsored by FutureQuest, Inc.
ossi@FutureQuest.net  http://www.FutureQuest.net/
-------------------------------------------------------------------------------
Changes in version 0.14

- Fixed compatibility with Python 2.6

- Fixed numbering of attachments.

Development of this version has been sponsored by FutureQuest, Inc.
ossi@FutureQuest.net  http://www.FutureQuest.net/
-------------------------------------------------------------------------------
Changes in version 0.13

- Added patch from John Simpson to allow masking email addresses to be
  turned off.

- Fixed handling archives with invalid month numbers.

- Fixed to work with Python 2.5.

Development of this version has been sponsored by FutureQuest, Inc.
ossi@FutureQuest.net  http://www.FutureQuest.net/
-------------------------------------------------------------------------------
Changes in version 0.12

- Moved all static files into a separate "files" directory.

- Fixed the distribution tarball to actuall include all source files.

- Fixed the previous "fix" to bogus month thread index filenames.

- Added a feed icon to the lists page from http://www.feedicons.com/

Note: The file move above will likely cause breakage on upgrades.  See
the UPGRADE file for more information.

Development of this version has been sponsored by FutureQuest, Inc.
ossi@FutureQuest.net  http://www.FutureQuest.net/
-------------------------------------------------------------------------------
Changes in version 0.11

- Fixed problem in handling of missing articles when searching, and
  missing month thread indexes.

- Fixed a problem with bogus month thread index filenames.

- Fixed several build issues, including problems finding the python
  program.

Development of this version has been sponsored by FutureQuest, Inc.
ossi@FutureQuest.net  http://www.FutureQuest.net/
-------------------------------------------------------------------------------
Changes in version 0.10

- Added support for generation of syndication feeds.

- Added support for reversing the order of the "by Date" message list.

- Added support for blocking or allowing for downloading raw messages.

- Added CSSPREFIX configuration item, to allow for easy configuration of
  where CSS files are located.  Thanks Roger Merchberger.

- Added support for word-wrapping message bodies (defaulting to off).
  Idea from Roger Merchberger.

Development of this version has been sponsored by FutureQuest, Inc.
ossi@FutureQuest.net  http://www.FutureQuest.net/
-------------------------------------------------------------------------------
Changes in version 0.9

- Two bugfixes in the handling of month numbers from "Central Linux
  Experts And Newbies" via Gerrit Pape.
- Some HTML template bugfixes.
-------------------------------------------------------------------------------
Changes in version 0.8

- Rewrote the front-end from scratch, separating content generation from
  the HTML templates.  This means less flexability in the templates, but
  faster code and easier maintenance and templating.
- Now has author indices, paging, more links, time zone management,
  style configuration, and other features I've forgotten.
- The default template is marked up with CSS classes in many places, and
  three sample style sheets are included.
-------------------------------------------------------------------------------
Changes in version 0.6

- Fixed a bug in EzmlmArchive.month.
- Added new built-in functions for producing relative links in the
  templates: relink and cmdlink.
- Rewrote the templates to merge all the common header and footer data
  into a single file.
- Added a parse tree optimizer to the template system which removes body
  elements consisting of a single node.
- Pre-compile all template expressions before use to speed up looped
  execution.
-------------------------------------------------------------------------------
Changes in version 0.5

- Totally rebuild the template system, moving most of the actual
  page-specific code into the template pages themselves.
-------------------------------------------------------------------------------
Changes in version 0.2

- Added proper targets to build the compiled Python bytecode files.

- Replaced the Python main routine with a C wrapper that is suitable for
  installations where setuid operation is required.
-------------------------------------------------------------------------------


ezmlm-browse-setup.py
import getopt
import os
import sys

argv0 = sys.argv[0]

def die_usage():
	print 'usage: %s [options] DESTPATH TARGET.cgi' % argv0
	print '  Options:'
	print '  --apache -- Create a .htaccess file for Apache'
	print '  --setuid -- Copy in a setuid wrapper program'
	sys.exit(1)

try:
	opts,args = getopt.getopt(sys.argv[1:], '',
							  [ 'apache', 'setuid' ])
	opts = dict(opts)
except getopt.GetoptError, x:
	print '%s: Invalid option: %s' % (argv0, x)
	die_usage()

if len(args) == 2:
	destbase,target = args
else:
	die_usage()

srcbase = os.getcwd()

def destpath(name):
	return os.path.join(destbase, name)

def srcpath(name):
	return os.path.join(srcbase, name)

def copyfile(srcname, destname=None, mode=None):
	data = file(srcpath(srcname)).read()
	dest = destpath(destname or srcname)
	file(dest, 'w').write(data)
	if mode:
		os.chmod(dest, mode)

if not os.path.isdir(destbase):
	os.mkdir(destbase, 0777)

try:
	os.unlink(destpath(target))
except:
	pass

if '--setuid' in opts:
	if os.getuid() == 0 or os.getgid() == 0:
		print '"%s: NEVER use the --setuid option as root' % argv0
		sys.exit(1)
	copyfile('wrapper', target, 06711)
else:
	copyfile('ezmlm-browse', target, 0755)

if not os.path.exists(destpath('ezmlm-browse.ini')):
	copyfile('ezmlm-browse.ini')

if '--apache' in opts:
	f = open(destpath('.htaccess'), 'w')
	f.write('DirectoryIndex %s\n'
			'AddHandler cgi-script .cgi\n'
			'Options +ExecCGI\n' % target)
	f.close()

print 'ezmlm-browse has been set up in "%s"' % destbase
print 'Make sure to edit "%s" for your lists' % destpath('ezmlm-browse.ini')
